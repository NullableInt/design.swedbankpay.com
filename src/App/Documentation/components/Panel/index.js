import React, { Component } from "react";

import { ComponentPreview, DocToc, Property } from "#";
import PanelComponent from "@/Panel";

const Overview = () => (
    <>
        <h2 id="overview">Overview</h2>
        <p>Panel...</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <PanelComponent
                title="Panel title"
                subTitle="Panel subtitle"
                footerText="Footer content goes here."
                footerBtnText="Footer button"
            >
                <p>Your main panel content is put here.</p>
                <p>Some more content over here.</p>
                <p>And one more line, just to be safe.</p>
            </PanelComponent>
        </ComponentPreview>
    </>
);

const PanelHeaders = () => (
    <>
        <h2 id="panel-headers">Panel headers</h2>
        <p>Two types of headers... <Property value=".panel-default" /> and <Property value=".panel-brand" />...</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <PanelComponent title="Default panel" />{"\n\n"}
            <PanelComponent type="brand" title="Panel brand" />
        </ComponentPreview>
    </>
);

const PanelDark = () => (
    <>
        <h2 id="panel-dark-mode">Panel dark mode</h2>
        <p>Dark body with <Property value=".panel-dark" />...</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <PanelComponent
                type="brand"
                title="Panel title"
                subTitle="Panel subtitle"
                footerText="Footer content goes here."
                footerBtnText="Footer button"
                darkMode
            >
                <p className="lead">This is a dark panel.</p>
                <p>Your main panel content is put here.</p>
                <p>Some more content over here.</p>
                <p>And one more line, just to be safe.</p>
            </PanelComponent>
        </ComponentPreview>
    </>
);

const PanelMuted = () => (
    <>
        <h2 id="muted-panel">Muted panel</h2>
        <p>Muted panel with <Property value=".panel-muted" />...</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <PanelComponent
                type="muted"
                title="Panel title"
                subTitle="Panel subtitle"
                footerText="Footer content goes here."
                footerBtnText="Footer button"
            >
                <p className="lead">This is a muted panel.</p>
                <p>Your main panel content is put here.</p>
                <p>Some more content over here.</p>
                <p>And one more line, just to be safe.</p>
            </PanelComponent>
        </ComponentPreview>
    </>
);

const PanelText = () => (
    <div className="col-lg-10 doc-body">
        <p className="lead">Panels...</p>
        <Overview />
        <PanelHeaders />
        <PanelDark />
        <PanelMuted />
    </div>
);

class Panel extends Component {
    render () {
        return (
            <div className="doc-container">
                <div className="row">
                    <PanelText />
                    <DocToc component={PanelText} />
                </div>
            </div>
        );
    }
}

export default Panel;

/* For testing */
export { Overview, PanelHeaders, PanelDark, PanelMuted, PanelText };
