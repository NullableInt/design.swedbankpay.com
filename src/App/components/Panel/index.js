import React from "react";
import PropTypes from "prop-types";

const Panel = ({ title, subTitle, id, type, removeBottomPadding, removeAllPadding, halfPadding, darkMode, children, footerText }) => {
    return (
        <section
            className={`panel panel-${type || "default"}${removeBottomPadding ? " panel-bottomless" : ""}${removeAllPadding ? " panel-no-padding" : ""}${halfPadding ? " panel-half-padding" : ""}${darkMode ? " panel-dark-mode" : ""}`}
            id={id}>
            {title ?
                <header>
                    <h2 className="panel-title">{title}</h2>
                    {subTitle ? <p className="panel-sub-title">{subTitle}</p> : null}
                </header>
                : null}
            {children ?
                <div className="panel-body">{children}</div>
                : null}
            {footerText ?
                <footer>
                    {footerText ? <p>{footerText}</p> : null}
                </footer>
                : null}
        </section>
    );
};

Panel.propTypes = {
    title: PropTypes.string,
    subTitle: PropTypes.string,
    id: PropTypes.string,
    type: PropTypes.oneOf(["brand", "default"]),
    halfPadding: PropTypes.bool,
    removeAllPadding: PropTypes.bool,
    removeBottomPadding: PropTypes.bool,
    darkMode: PropTypes.bool,
    footerText: PropTypes.string
};

export default Panel;
