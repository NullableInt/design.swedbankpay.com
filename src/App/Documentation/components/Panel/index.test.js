import React from "react";
import { shallow } from "enzyme";

import Panel from "./index";
import { Overview, PanelHeaders, DarkMode, PanelMuted, PanelText } from "./index";

describe("Documentation: Panel", () => {
    it("is defined", () => {
        expect(Panel).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(
            <Panel />
        );

        expect(wrapper).toMatchSnapshot();
    });

    describe("Overview", () => {
        it("is defined", () => {
            expect(Overview).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(
                <Overview />
            );

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("PanelHeaders", () => {
        it("is defined", () => {
            expect(PanelHeaders).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(
                <PanelHeaders />
            );

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("DarkMode", () => {
        it("is defined", () => {
            expect(DarkMode).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(
                <DarkMode />
            );

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("PanelMuted", () => {
        it("is defined", () => {
            expect(PanelMuted).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(
                <PanelMuted />
            );

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("PanelText", () => {
        it("is defined", () => {
            expect(PanelText).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(
                <PanelText />
            );

            expect(wrapper).toMatchSnapshot();
        });
    });
});
