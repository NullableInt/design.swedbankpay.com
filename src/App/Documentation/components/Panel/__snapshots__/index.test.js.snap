// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documentation: Panel Overview renders 1`] = `
<Fragment>
  <h2
    id="overview"
  >
    Overview
  </h2>
  <p>
    Panel...
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Panel
      footerBtnText="Footer button"
      footerText="Footer content goes here."
      subTitle="Panel subtitle"
      title="Panel title"
    >
      <p>
        Your main panel content is put here.
      </p>
      <p>
        Some more content over here.
      </p>
      <p>
        And one more line, just to be safe.
      </p>
    </Panel>
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: Panel PanelDark renders 1`] = `
<Fragment>
  <h2
    id="panel-dark-mode"
  >
    Panel dark mode
  </h2>
  <p>
    Dark body with 
    <Property
      value=".panel-dark"
    />
    ...
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Panel
      darkMode={true}
      footerBtnText="Footer button"
      footerText="Footer content goes here."
      subTitle="Panel subtitle"
      title="Panel title"
      type="brand"
    >
      <p
        className="lead"
      >
        This is a dark panel.
      </p>
      <p>
        Your main panel content is put here.
      </p>
      <p>
        Some more content over here.
      </p>
      <p>
        And one more line, just to be safe.
      </p>
    </Panel>
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: Panel PanelHeaders renders 1`] = `
<Fragment>
  <h2
    id="panel-headers"
  >
    Panel headers
  </h2>
  <p>
    Two types of headers... 
    <Property
      value=".panel-default"
    />
     and 
    <Property
      value=".panel-brand"
    />
    ...
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Panel
      title="Default panel"
    />
    


    <Panel
      title="Panel brand"
      type="brand"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: Panel PanelMuted renders 1`] = `
<Fragment>
  <h2
    id="muted-panel"
  >
    Muted panel
  </h2>
  <p>
    Muted panel with 
    <Property
      value=".panel-muted"
    />
    ...
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Panel
      footerBtnText="Footer button"
      footerText="Footer content goes here."
      subTitle="Panel subtitle"
      title="Panel title"
      type="muted"
    >
      <p
        className="lead"
      >
        This is a muted panel.
      </p>
      <p>
        Your main panel content is put here.
      </p>
      <p>
        Some more content over here.
      </p>
      <p>
        And one more line, just to be safe.
      </p>
    </Panel>
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: Panel PanelText renders 1`] = `
<div
  className="col-lg-10 doc-body"
>
  <p
    className="lead"
  >
    Panels...
  </p>
  <Overview />
  <PanelHeaders />
  <PanelDark />
  <PanelMuted />
</div>
`;

exports[`Documentation: Panel renders 1`] = `
<div
  className="doc-container"
>
  <div
    className="row"
  >
    <PanelText />
    <DocToc
      component={[Function]}
    />
  </div>
</div>
`;
